libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteSetValue.kt

```diff
@@ -22,10 +20,8 @@ class RouteSetValue internal constructor(
     val trafficLineExpressionProvider: RouteLineExpressionProvider?,
     val routeLineExpression: Expression,
     val casingLineExpression: Expression,
-    val altRoute1TrafficExpressionProvider: RouteLineExpressionProvider?,
-    val altRoute2TrafficExpressionProvider: RouteLineExpressionProvider?,
-    val alternativeRoute1Source: FeatureCollection,
-    val alternativeRoute2Source: FeatureCollection,
+    val alternativeRoutesTrafficExpressionProviders: List<RouteLineExpressionProvider?>,
```

Comment:
The `MapboxRouteLineView` is needing to know which item in the list corresponds to which layer ID.  This looks error prone going forward.  Could we instead use an object that combine the layerID with the `RouteLineExpressionProvider` so that the view class doesn't have to have too much knowledge?

Something like a list of:
data class SomeObject(layerID: String, RouteLineExpressionProvider)

Reply:
The Layer ID is an implementation detail of the View, the API doesn't need to know that the View is using string IDs to add geometries or expressions to the map. This is going to become especially true when we migrate away from using constants as layer IDs to generate them on the fly depending on how many alternatives there are going to be.

What I thought that we could join together is the geometry with the expression. These are separate lists now but it should rather be a single list of:
```
SomeObject(FeatureCollection, RouteLineExpressionProvider)
```

Reply:
`MapboxRouteLineView` line 136: `value.alternativeRouteLinesData[0].also {...`    This is what I have reservations about.  The view class has to trust that the element at index 0 corresponds with `ALTERNATIVE_ROUTE_1_SOURCE_ID`.  There's nothing that enforces this relationship.  The idea of the view classes has been that there's no business logic so to me it makes sense that the API class is packaging up the data and the view class just renders it.  So the API class knows which `RouteLineData` goes with which layer source. It seems natural to me that the API class would package the data that makes that relationship explicit such that the view class renders the data that's packaged together rather than assuming or trusting that the ordering of the data in the collection(s) it receives has some significance.

Reply:
I see what you're saying and this is something we might want to do, especially if we start generating the IDs for multiple alternative routes.

Layer IDs are an internal API though, so we can refactor it at any point later and I'd like to avoid adding even more changes here if possible.

With the current refactor we access the index only once, so the data will always be rendered correctly, it might be on different layers which is probably a low-risk problem.